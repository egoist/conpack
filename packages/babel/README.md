# @conpack/babel

## Install

```bash
yarn add @conpack/babel
```

## Usage

```js
const Conpack = require('conpack')
const babel = require('@conpack/babel')

const config = new Conpack()
babel(config, options)

module.exports = config.toConfig()
```

## API

### options

#### options.babelLoaderOptions

Type: `any`<br>
Default: `undefined`

Options for `babel-loader`.

## License

MIT &copy; [EGOIST](https://github.com/egoist)
