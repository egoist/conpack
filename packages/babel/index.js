module.exports = (config, { babelLoaderOptions } = {}) => {
  const jsRule = config.rules.add('js', {
    test: /\.jsx?$/,
    exclude: [/node_modules/]
  })

  jsRule.loaders.add('babel-loader', {
    loader: 'babel-loader',
    options: babelLoaderOptions
  })

  return config
}
