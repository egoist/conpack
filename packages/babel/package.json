{
  "name": "@conpack/babel",
  "version": "0.0.0",
  "publishConfig": {
    "access": "public"
  },
  "files": [
    "index.js"
  ],
  "license": "MIT",
  "devDependencies": {
    "conpack": "^0.3.7"
  },
  "xo": false
}
