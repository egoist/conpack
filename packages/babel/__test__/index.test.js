import Conpack from 'conpack'
import babel from '..'

describe('main', () => {
  it('defaults', () => {
    const config = new Conpack()
    babel(config)
    expect(config.toConfig()).toMatchSnapshot()
  })

  it('babel-loader options', () => {
    const config = new Conpack()
    babel(config, {
      babelLoaderOptions: {
        babelrc: false
      }
    })
    expect(config.toConfig()).toMatchSnapshot()
  })
})
