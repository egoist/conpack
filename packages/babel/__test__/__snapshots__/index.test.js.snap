// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main babel-loader options 1`] = `
Object {
  "module": Object {
    "rules": Array [
      Object {
        "exclude": Array [
          /node_modules/,
        ],
        "test": /\\\\\\.jsx\\?\\$/,
        "use": Array [
          Object {
            "loader": "babel-loader",
            "options": Object {
              "babelrc": false,
            },
          },
        ],
      },
    ],
  },
  "plugins": Array [],
}
`;

exports[`main defaults 1`] = `
Object {
  "module": Object {
    "rules": Array [
      Object {
        "exclude": Array [
          /node_modules/,
        ],
        "test": /\\\\\\.jsx\\?\\$/,
        "use": Array [
          Object {
            "loader": "babel-loader",
            "options": undefined,
          },
        ],
      },
    ],
  },
  "plugins": Array [],
}
`;
